src/mscorlib/shared/System/Globalization/TextInfo.cs

```diff
@@ -809,11 +809,21 @@ private int AddTitlecaseLetter(ref StringBuilder result, ref string input, int i
         {
             Debug.Assert(charLen == 1 || charLen == 2, "[TextInfo.AddTitlecaseLetter] CharUnicodeInfo.InternalGetUnicodeCategory returned an unexpected charLen!");
 
-            // for surrogate pairs do a simple ToUpper operation on the substring
+            // for surrogate pairs do a ToUpper operation on the substring
             if (charLen == 2)
             {
                 // Surrogate pair
-                result.Append(ToUpper(input.Substring(inputIndex, charLen)));
+                ReadOnlySpan<char> src = input.AsSpan(inputIndex, 2);
+                Span<char> dst = stackalloc char[2];
+                if (GlobalizationMode.Invariant)
+                {
+                    ToUpperAsciiInvariant(src, dst);
```

Comment:
>ToUpperAsciiInvariant(src, dst); [](start = 20, length = 32)

I don't think we need to do that as we are in invariant mode and we know we are handling Surrogate. we can just append the original characters as these characters will be outside the ascii range anyway,

Reply:
So you mean change this to something like:
```C#
if (_invariantMode)
{
    result.Append(src);
}
else
{
    Span<char> dst = stackalloc char[2];
    ChangeCase(src, dst, toUpper: true);
    result.Append(dst);
}
```
?

Reply:
Yes.